server-common/src/main/java/org/apache/kafka/queue/KafkaEventQueue.java

```diff
@@ -250,7 +250,7 @@ private void handleEvents() throws InterruptedException {
                             continue;
                         } else if (shuttingDown) {
                             remove(eventContext);
-                            toDeliver = new TimeoutException();
+                            toDeliver = new QueueClosingException();
```

Comment:
Hmm, I'm not really sure about having a new execption here. Can we just use `RejectedExecutionException` here?

Reply:
Also we will need a test in `KafkaEventQueueTest` if you want to do this. I do wonder if this should be a separate PR ... queue changes have a way of being big. But I guess let's see if it impacts any other tests.

Reply:
I thought about `RejectedExecutionException`, but wasn't sure if we relied on the `TimeoutException` type somewhere, so I made a subclass of that. If you think it's fine, we'll go with `RejectedExecutionException`.

Reply:
I think `RejectedExecutionException` really is more appropriate. If you tried to schedule (or reschedule) a deferred task and the queue was shut down, you'd get `RejectedExecutionException` already. So you already have to catch REE and treat it this way.